BuiltDifferentMobileApp/BuiltDifferentMobileApp/ViewModels/Profile/MyProfilePageClientViewModel.cs

```diff
@@ -68,19 +68,24 @@ public bool IsEnabled
         private INetworkService<HttpResponseMessage> networkService = NetworkService<HttpResponseMessage>.Instance;
 
         public MyProfilePageClientViewModel() {
+            GetUserInfo();
+
+            SubmitCommand = new AsyncCommand(Submit);
+            UploadImageCommand = new AsyncCommand(Upload);
+            EditProfileCommand = new AsyncCommand(Edit);
+
+            isEnabled = false;
+        }
+
+        private async Task GetUserInfo()
+        {
             Models.Client user = (Models.Client)accountService.CurrentUser;
             Id = user.id;
             Name = user.name;
             UserId = user.userId;
             ProfilePicture = user.profilePicture;
             StartWeight = user.startWeight;
             CurrentWeight = user.currentWeight;
-
-            SubmitCommand = new AsyncCommand(Submit);
-            UploadImageCommand = new AsyncCommand(Upload);
-            EditProfileCommand = new AsyncCommand(Edit);
-
-            isEnabled = false;
         }
 
         private async Task Edit()
```

Comment:
Everything looks good, good job!
While we're here, I'm just wondering for the profile there is an edit button and you can click it to switch between edit and non-edit mode, correct?

Reply:
yes correct, ill double check to make sure its in working order

Reply:
Yes and there's a confirm/cancel button correct?
If so, be careful about user's inputting changes and cancelling their changes, you will need to set the form's fields to their previous values.

Reply:
Changes have been made for this suggestion, will reretrieve the profile if not saved. (reseting the profile data